apps/emqx_bridge_azure_blob_storage/src/emqx_bridge_azure_blob_storage_action_schema.erl

```diff
@@ -0,0 +1,310 @@
+        mk(
+            mkunion(mode, #{
+                <<"direct">> => ref(direct_parameters),
+                <<"aggregated">> => ref(aggreg_parameters)
+            }),
+            #{
+                required => true,
+                desc => ?DESC("parameters")
+            }
+        ),
+        #{
+            resource_opts_ref => ref(action_resource_opts)
+        }
+    );
+fields(direct_parameters) ->
+    [
+        {mode, mk(direct, #{required => true, desc => ?DESC("direct_mode")})},
+        %% Container in the Azure Blob Storage domain, not aggregation.
+        {container,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("direct_container_template"),
+                    required => true
+                }
+            )},
+        {blob,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("direct_blob_template"),
+                    required => true
+                }
+            )},
+        {content,
+            mk(
+                emqx_schema:template(),
+                #{
+                    required => false,
+                    default => <<"${.}">>,
+                    desc => ?DESC("direct_content_template")
+                }
+            )}
+        | fields(common_action_parameters)
+    ];
+fields(aggreg_parameters) ->
+    [
+        {mode, mk(aggregated, #{required => true, desc => ?DESC("aggregated_mode")})},
+        {aggregation, mk(ref(aggregation), #{required => true, desc => ?DESC("aggregation")})},
+        %% Container in the Azure Blob Storage domain, not aggregation.
+        {container,
+            mk(
+                string(),
+                #{
+                    desc => ?DESC("aggregated_container_name"),
+                    required => true
+                }
+            )},
+        {blob,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("aggregated_blob_template"),
+                    required => true
+                }
+            )}
+        | fields(common_action_parameters)
+    ];
+fields(aggregation) ->
+    [
+        emqx_connector_aggregator_schema:container(),
+        %% TODO: Needs bucketing? (e.g. messages falling in this 1h interval)
+        {time_interval,
+            hoconsc:mk(
+                emqx_schema:duration_s(),
+                #{
+                    required => false,
+                    default => <<"1h">>,
+                    desc => ?DESC("aggregation_interval")
+                }
+            )},
+        {max_records,
+            hoconsc:mk(
+                pos_integer(),
+                #{
+                    required => false,
+                    default => 1_000_000,
+                    desc => ?DESC("aggregation_max_records")
+                }
+            )},
+        {min_block_size,
```

Comment:
I feel this is also a bit out of place here in `aggregation` options. 🌝

Reply:
Well, this one really only affects the aggregation part of the `aggregated` mode, differently from `max_block_size` which indeed meant Azure's API limits.  :thinking:

Reply:
I get your point, though I'd still think about the delivery process as being separate from the aggregation process.

Reply:
I see.  I've moved it one level up here: https://github.com/emqx/emqx/pull/13069/commits/34d8432db0761003a386eb6c80da8e765fca0507